sympy/concrete/tests/test_sums_products.py

```diff
@@ -1603,6 +1604,30 @@ def test_summation_by_residues():
     assert eval_sum_residue((-1)**x / x**2, (x, S(1), oo)) == -pi**2/12
     assert eval_sum_residue((-1)**x / x**2, (x, S(2), oo)) == 1 - pi**2/12
 
+    # https://github.com/sympy/sympy/issues/27824
+    #An example of even function which works for -oo to k
+    assert eval_sum_residue((1/(k**2+1)), (k, -oo, S(0))) == Rational(1, 2) + pi/(2*tanh(pi))
+
+    #An example of function which is neither even nor odd which works for -oo to k
+    assert eval_sum_residue(1 / (k**2 + 2*k +2), (k, -oo, S(0))) == 1 + pi/(2*tanh(pi))
+    ans = Rational(1, 2) + pi/(2*tanh(pi))
+    assert eval_sum_residue((1/(k**2+1)), (k, -oo, S(0))) == ans
+    assert eval_sum_residue((1/(k**2+1)), (k, oo, -S(1))) == -ans  # Karr convention
+
+    # Odd functions over (-oo to k) are handled by other parts of the summation
+    # logic, so eval_sum_residue will return None in such cases.
+    # The odd function when called directly in eval_sum_residue returns None
+    # because the current implementation attempts to convert non-even functions
+    # into even ones by shifting. However, this doesn't work in this case,
+    # as the coefficient of the second-highest power in the denominator (k**2)
+    # is zero. This results in a shift of zero, which means the function should
+    # return None.
+    assert eval_sum_residue((1/k**3), (k,-oo, S(-1))) == None
+    # SO issue cited on #27827
+    # this will return 0 if it is not allowed to shift to become even
```

Comment:
Are these comments still true?

I don't think we need the lengthy first comment in any case.